### PR TITLE
Extract click-behavior e2e reproductions

### PR DESCRIPTION
**Pull request by kamilmielnik**
### Description

First step in #31498.
Lots of new tests will be added to `e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js` so I'm moving bug reproductions out of that file to make it easier to work with once it starts to grow.

### CODE REVIEW

File: e2e/test/scenarios/dashboard-cards/reproductions/16334-click-behavior-question-with-filter-changes-viz-type.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: { id, card_id, dashboard_id } }) => {
+          addOrUpdateDashboardCard({
+            dashboard_id,
+            card_id,
+            card: {
+              id,
+              visualization_settings: getVisualizationSettings(question1Id),
+            },
+          });
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard-cards/reproductions/16334-click-behavior-question-with-filter-changes-viz-type.cy.spec.js:42]: I also reused `addOrUpdateDashboardCard` here, see b511b8b6b420fa18ac35a2880b44402a0f6e4625

File: e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    });
+
+    cy.findByTestId("progress-bar").click();
+    cy.wait("@dataset");
+    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+    cy.findByText("Orders");
+  });
+});
+
+const getQuestionDetails = ({ display }) => ({
```
Comment (WiNloSt) [inline e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js:53]: I'd just create 2 top-level variables maybe
- `GAUGE_QUESTION_DETAILs` and
- `PROGRESS_QUESTION_DETAILS`
Or other names as you see fit as we usually declare our dashboard/question details in the top of the files. And there isn't much duplication here at all. Declaring them at the top would be more consistent.

**Review by WiNloSt — DISMISSED** [review]
Looks good to me

File: e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js (hunk 2 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js]: 77b5a1fe671686720a9e8b7f8e34e6cf3b90e782

File: e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  display,
+  query: {
+    "source-table": REVIEWS_ID,
+    aggregation: [["count"]],
+  },
+});
+
+const getDashcardDetails = ({ id, card_id, target_id }) => ({
```
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js]: That works :+1: 8518d5487acbbe8c7e97bacbb8d3a64a85553870

**Comment by kamilmielnik** [comment]
image

This is... weird... I just pushed a commit.

@WiNloSt can you please have another look?

**Review by WiNloSt — APPROVED** [review]
The last 2 commits look good to me.

**Pull request merged**